src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -558,6 +578,21 @@ async function InitializeFirebaseAdminStubs(frb: FunctionsRuntimeBundle): Promis
       databaseApp = adminModuleTarget.initializeApp(databaseAppOptions, DATABASE_APP);
       proxiedDatabase = makeProxiedDatabase(adminModuleTarget);
 
+      // Tell the Firebase Functions SDK to use the proxied app so that things like "change.after.ref"
+      // point to the right place.
+      const functionsVersion = parseVersionString(functionsResolution.version);
+      if (functionsVersion.major >= 3 && functionsVersion.minor >= 3) {
```

Comment:
The current check will fail for `4.1` (since minor is less than 3). Please consider using feature detection like `if (localFunctionsModel.app.setEmulatedAdminApp)` instead of checking version.

Reply:
Good call.